algorithms/src/snark/varuna/ahp/indexer/indexer.rs

```diff
@@ -228,23 +228,21 @@ impl<F: PrimeField, SM: SNARKMode> AHPForR1CS<F, SM> {
         state: IndexerState<F>,
         id: &CircuitId,
         point: F,
-    ) -> Result<impl Iterator<Item = F>, AHPError> {
-        let labels = [["a"], ["b"], ["c"]];
-        let mut evals = [
-            (state.a_arith, state.non_zero_a_domain),
-            (state.b_arith, state.non_zero_b_domain),
-            (state.c_arith, state.non_zero_c_domain),
-        ]
-        .into_iter()
-        .zip_eq(&labels)
-        .flat_map(|((evals, domain), label)| {
+    ) -> Result<impl ExactSizeIterator<Item = F>> {
+        let mut all_evals = Vec::with_capacity(12);
```

Comment:
@ljedrz allocation amount needs further review. Can you validate size?

Reply:
this one is most likely fine, as we've used `collect` before; I'll run some benches to double-check